src/Invio.Immutable/SingleComparisonAttribute.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+
+namespace Invio.Immutable {
+    /// <summary>
+    ///   Indicates that the decorrated property should be compared based on a specified level of
+    ///   precision.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class SingleComparisonAttribute : Attribute {
+        /// <summary>
+        ///   The level of precision to use when comparing floating point values.
+        /// </summary>
+        public Int32 Precision { get; }
+
+        /// <summary>
+        ///   Indicates whether precision should be interpreted as the number of signficant figures,
+        ///   or as the number of signficants digits to the right of the decimal place.
+        /// </summary>
+        public Boolean SignificantFigures { get; set; }
```

Comment:
Consider making this an enum instead of a boolean to be more definitive in what it is doing. My rationale is that this property does not represent the existence of (or lack of) significant figures, but how the precision will be applied. I'm not loving the name "style", but here's an example of where my brain's at:

```
public class MyImmutable : ImmutableBase<Foo> {

    [DoubleComparison(precision: 10, style: PrecisionStyle.SignificantFigures)]
    public Double MyDoubleProperty { get; }

    [DoubleComparison(precision: 5, style: PrecisionStyle.DecimalPlaces)]
    public Double MyOtherProperty { get; }
}
```